Sources/MongoSwift/Operations/CreateCollectionOperation.swift

```diff
@@ -132,15 +132,12 @@ internal struct CreateCollectionOperation<T: Codable>: Operation {
             self.database._database, self.name, opts?.data, &error) else {
             throw parseMongocError(error)
         }
+        mongoc_collection_destroy(collection)
```

Comment:
this place was a little tricky, because previously we just passed the `mongoc_collection_t` that `mongoc_database_create_collection` returns into the initializer.

I rewrote this so we just throw away that instance and pass on the relevant info to the initializer. 
I considered having a second `MongoCollection` initializer that would still use the pointer, but I think once we add client pooling we will throw away the collection here anyway, so I figured it wasn't worth it.